### PR TITLE
Change `Successors` to `impl Iterator<Item = BasicBlock>`

### PR DESCRIPTION
**Pull request by SparrowLii**
This PR fixes the FIXME in `compiler\rustc_middle\src\mir\mod.rs`.
This can omit several `&`, `*` or `cloned` operations on Successros' generated elements

### CODE REVIEW

File: compiler/rustc_mir_transform/src/coverage/graph.rs [hunk 1]
```diff
@@ -9,6 +9,7 @@ use rustc_index::vec::IndexVec;
 use rustc_middle::mir::coverage::*;
 use rustc_middle::mir::{self, BasicBlock, BasicBlockData, Terminator, TerminatorKind};
 
+use rustc_middle::mir::TerminatorKind::SwitchInt;
```
Comment (lcnr) [inline compiler/rustc_mir_transform/src/coverage/graph.rs:12]: please revert this change to the previous state. Don't think that this is an improvement.

**Review by lcnr — COMMENTED** [review]
please revert the unrelated change. Apart from that this is great :+1: thanks

File: compiler/rustc_mir_transform/src/coverage/graph.rs (hunk 1 above)
Comment (SparrowLii) [reply compiler/rustc_mir_transform/src/coverage/graph.rs]: Thanks! Has fixed it
